Release 1.0.0

ztybigcat opened the pull request:
Updated README to a nicer format and linking to EC2 docker hosting guide.

ryangawei committed:
Merge pull request #23 from YuhanYue/22-reorganize-component-structure-add-pages-folder

ryangawei committed:
Add router, move RecordPage into `pages/`

ryangawei committed:
Merge pull request #24 from YuhanYue/21-add-react-router-to-frontend

Add router, move RecordPage into `pages/`

ryangawei committed:
Add handler for `get_scores`

ryangawei committed:
Add handler for `get_scores`

ryangawei committed:
Add handler for `get_scores`

codeeverydayyu committed:
Connect to mysql in EC2 and fetch reference table by route '/prompts'.

codeeverydayyu committed:
Connect to mysql in EC2 and fetch reference table by route '/prompts'.

codeeverydayyu committed:
Add version number to requirements. Move get_data to speak.py but have error.

codeeverydayyu committed:
Add version number to requirements. Move get_data to speak.py but have error.

YuhanYue committed:
Update backend set up requirement

ryangawei committed:
Update dependencies (webrtcvad for win, werkzeug)

codeeverydayyu committed:
working on import db connection

codeeverydayyu committed:
working on import db connection

ryangawei committed:
Merge pull request #30 from YuhanYue/req

Update Python version to 3.9.4 and dependencies

ryangawei committed:
Add record page in MUI

codeeverydayyu committed:
Fix moving db connection to db_helper and add library version.

codeeverydayyu committed:
Fix moving db connection to db_helper and add library version.

codeeverydayyu committed:
Add tests of /get_reference api and reference schema.

codeeverydayyu committed:
Add tests of /get_reference api and reference schema.

codeeverydayyu committed:
Change db to 'voice_collector'.

codeeverydayyu committed:
Change db to 'voice_collector'.

codeeverydayyu committed:
Merge branch 'dev' into 28-database-create-mysql-in-aws-ec2-and-connect-it-to-backend

YuhanYue committed:
Length, silence and volume check

codeeverydayyu committed:
Merge branch '28-database-create-mysql-in-aws-ec2-and-connect-it-to-backend' of https://github.com/YuhanYue/VoiceCollector into 28-database-create-mysql-in-aws-ec2-and-connect-it-to-backend

codeeverydayyu committed:
Merge requirements-dve.txt

codeeverydayyu committed:
Write record from back-end to database.

YuhanYue committed:
Update config.json and format check

codeeverydayyu committed:
Refactor the write_record_route() and add a unit test.

YuhanYue committed:
remove secret key

ryangawei committed:
Update dependencies

ryangawei committed:
Update .gitignore

ryangawei committed:
Add dummy data into record page

ryangawei committed:
Add MUI recorder page into router

ryangawei committed:
Remove GET methods in validate api

ryangawei committed:
Add wav format check

ryangawei committed:
Update major format check

ryangawei committed:
Add test cases and samples for format check

ryangawei committed:
Update filename check

ryangawei committed:
Update validate response with better msg

ryangawei committed:
Round duration to 2 decimals

ryangawei committed:
Fix silence check bugs

ryangawei committed:
Add default API test

ryangawei committed:
Add test cases for validate

ryangawei committed:
Update silence check name

ryangawei committed:
Change major format config

ryangawei committed:
Update gitignore

codeeverydayyu committed:
Add get_all_user_id route. Connect to S3, add write_file route.

codeeverydayyu committed:
Delete get_all_user_id route. Add exists_user_id route.

ryangawei committed:
Delete package-lock.json

ryangawei committed:
Delete RecordMUI.js

ryangawei committed:
Revert "Delete RecordMUI.js"

This reverts commit c24bd5951b03f0ce142ec93d2cba8b286be57054.

ryangawei committed:
Delete yarn.lock

ryangawei committed:
Merge branch '25-implement-intel-score-handler' of https://github.com/YuhanYue/VoiceCollector into 25-implement-intel-score-handler

ryangawei committed:
Merge pull request #34 from YuhanYue/20-refactor-recording-page-using-mui-containers-and-components

Add record page in MUI

ryangawei committed:
Merge branch 'dev' into 19-implement-silence-and-volume-checck

ryangawei committed:
Merge pull request #32 from YuhanYue/19-implement-silence-and-volume-checck

Length, silence and volume check

ryangawei committed:
Update intel handler

ryangawei committed:
Remove comment

ryangawei committed:
Add test for intel handler

ryangawei committed:
Merge branch '25-implement-intel-score-handler' of https://github.com/YuhanYue/VoiceCollector into 25-implement-intel-score-handler

ryangawei committed:
Update intel test

ryangawei committed:
Merge pull request #26 from YuhanYue/25-implement-intel-score-handler

Add handlers for intel scores

YuhanYue committed:
Fix vad_mode from config

ryangawei committed:
Merge pull request #40 from YuhanYue/37-fix-vad_mode-bug

Fix vad_mode from config

ryangawei committed:
Add recording component from Common Voice

ryangawei committed:
Add recording component to pages

YuhanYue committed:
Integrate audio recorder to recording page

and force one initialization of each audio recorder

YuhanYue committed:
Connect frontend with backend

codeeverydayyu committed:
Modify write_file_route and add test.

ryangawei committed:
Rename audio state

codeeverydayyu committed:
Modify write_record and its test.

codeeverydayyu committed:
Delete get user id route.

YuhanYue committed:
Merge pull request #42 from YuhanYue/15-add-audio-recording-component-into-recording-page

15 add audio recording component into recording page

YuhanYue committed:
Add format convert api

ryangawei committed:
Revert "Add format convert api"

This reverts commit b60ce76178392bfd07afbf384e57feb7b135b445.

ryangawei committed:
Update recorder to output wav

ryangawei committed:
Change record output format into webm

ryangawei committed:
Convert from webm to wav

ryangawei committed:
Minor fix

ryangawei committed:
Change config back

ryangawei committed:
Rename check in validate

ryangawei committed:
Fix change format code

ryangawei committed:
Add read audio test script

ryangawei committed:
Response with converted file

ryangawei committed:
Update README

YuhanYue committed:
Merge pull request #44 from YuhanYue/43-convert-audio-file-format

Add convert format handler, change recorder format to `webm`

ryangawei committed:
Merge branch 'dev' into 28-database-create-mysql-in-aws-ec2-and-connect-it-to-backend

ryangawei committed:
Merge pull request #29 from YuhanYue/28-database-create-mysql-in-aws-ec2-and-connect-it-to-backend

Connect to database.

ryangawei committed:
Add script to infer scores from TORGO

ryangawei committed:
Merge pull request #46 from YuhanYue/41-compute-metrics-using-participant-severity-and-automatic-intel-scores

Add script to infer scores from TORGO

ryangawei committed:
Change validate handler structure

ryangawei committed:
Change validate handler structure

ryangawei committed:
Change validate handler structure

ryangawei committed:
Change validate handler structure

ryangawei committed:
Change form-data key to "audio", add error code

ryangawei committed:
Change form-data key to "audio", add error code

ryangawei committed:
Change form-data key to "audio", add error code

ryangawei committed:
Update intel handler

ryangawei committed:
Update intel handler

ryangawei committed:
Update intel handler

codeeverydayyu committed:
Delete audio_id in write record methods.

ryangawei committed:
Merge pull request #48 from YuhanYue/modify-write-record

Delete audio_id in write record methods.

ryangawei committed:
Merge branch '47-refactor-handler-code-to-support-validation-pipeline' of https://github.com/YuhanYue/VoiceCollector into 47-refactor-handler-code-to-support-validation-pipeline

ryangawei committed:
Change upload file key to audio in speak api

ryangawei committed:
Change upload file key to audio in speak api

ryangawei committed:
Add test for format api

ryangawei committed:
Add test for format api

ryangawei committed:
Change key name in format api

ryangawei committed:
Change key name in format api

ryangawei committed:
Format api handler wav input

ryangawei committed:
Format api handler wav input

ryangawei committed:
Format api also accept wav

ryangawei committed:
Format api also accept wav

YuhanYue committed:
Submit button and navigation

codeeverydayyu committed:
Checked record format before call sql statement and added test.

ryangawei committed:
Merge pull request #51 from YuhanYue/modify-write-file

Checked record format before call sql statement and added test.

ryangawei committed:
Remove global var

ryangawei committed:
Remove global var

ryangawei committed:
Add get reference and record helper

ryangawei committed:
Add get reference and record helper

ryangawei committed:
Add submit handler, get reference and records

ryangawei committed:
Add submit handler, get reference and records

ryangawei committed:
Add test for submit

ryangawei committed:
Add test for submit

ryangawei committed:
Reorganize code

ryangawei committed:
Reorganize code

ryangawei committed:
Merge branch '47-refactor-handler-code-to-support-validation-pipeline' of https://github.com/YuhanYue/VoiceCollector into 47-refactor-handler-code-to-support-validation-pipeline

ryangawei committed:
Change test data key

ryangawei committed:
Change test data key

codeeverydayyu committed:
add db thread lock; wrap get_reference with try catch; read  prompt from db.

ryangawei committed:
Merge pull request #50 from YuhanYue/49-frontend-submit-button-and-navigation

Submit button and navigation

codeeverydayyu committed:
Merge branch 'get-reference-from-db' into dev

codeeverydayyu committed:
Show image prompt.

YuhanYue committed:
Merge branch '47-refactor-handler-code-to-support-validation-pipeline' of https://github.com/YuhanYue/VoiceCollector into 47-refactor-handler-code-to-support-validation-pipeline

YuhanYue committed:
Add webm format

codeeverydayyu committed:
Modify reference_schema in test_api.

codeeverydayyu committed:
Modify reference_schema in test_api.

YuhanYue committed:
Form data and send to submission pipeline

codeeverydayyu committed:
Add image_url in reference_schema test.

codeeverydayyu committed:
Modify image section based on changes in mysql.

ryangawei committed:
Merge pull request #52 from YuhanYue/47-refactor-handler-code-to-support-validation-pipeline

47 refactor handler code to support validation pipeline

ryangawei committed:
Merge pull request #57 from YuhanYue/56-frontend-connect-to-validation-pipeline

Form data and send to submission pipeline

ryangawei committed:
Merge pull request #55 from YuhanYue/54-frontend-image-section

Show image prompt.

ryangawei committed:
Fix convert handler not recognize webm error

ryangawei committed:
Add get_records handler for better access all audio records.

ryangawei committed:
Clear blob when switching prompts, set upload date to mysql format

YuhanYue committed:
Add alert when submission is failed

ryangawei committed:
Merge pull request #62 from YuhanYue/61-frontend-pop-out-alert-when-audio-submission-is-failed

Add alert when submission is failed

ryangawei committed:
Merge branch 'dev' into clear-audio

ryangawei committed:
Change space

YuhanYue committed:
Only allow submit when there is audio recorded

ryangawei committed:
Merge pull request #60 from YuhanYue/clear-audio

Clear audio after switching prompt, fix date format

ryangawei committed:
Merge branch 'dev' into 61-frontend-pop-out-alert-when-audio-submission-is-failed

ryangawei committed:
Revert badly rebase sections

ryangawei committed:
Fix record button location changing issue when recording

ryangawei committed:
Change button color when submit successfully. Add alert transition

codeeverydayyu committed:
Merge pull request #64 from YuhanYue/61-frontend-pop-out-alert-when-audio-submission-is-failed

Only allow submit when there is audio recorded

ryangawei committed:
Remove particle effect in home page

ryangawei committed:
Block submit after success, add circle progress bar

ryangawei committed:
Merge pull request #66 from YuhanYue/remove-particle-add-progress-status

Remove particle add progress status

Commit:
Ticket 3 initial commit

Commit:
Ticket 3 initial commit

Commit:
initial commit

Commit:
updated comments from Aanchan

jordan-git-lewis committed:
Merge pull request #4 from SlangLab-NU/3-remove-localhost-url-and-make-the-ipdomain-configurable

3 remove localhost url and make the ipdomain configurable

Commit:
initial commit

Commit:
june 14 commit

Commit:
Fixed PR comments

Commit:
lock conflict check

Commit:
removed tmp files

Commit:
Merge branch 'dev' into 2-add-a-sql-database-for-running-app-locally

Commit:
moved file names to config

Commit:
merge db connection and add TYPE config variable

Commit:
merge db connection and write records

jordan-git-lewis committed:
Merge pull request #9 from SlangLab-NU/2-add-a-sql-database-for-running-app-locally

2 add a sql database for running app locally

Rao2321 committed:
added logger to the server and updated print statements to info level logs to console output

macarious committed:
Fix import statements

macarious committed:
Include logging_config.yaml in logger file

macarious committed:
Include logging_config.yaml in logger file

macarious committed:
Fix imports so both db_init.py and flask will both run properly

macarious committed:
Remove redundant print statement

macarious committed:
Merge pull request #17 from SlangLab-NU/logger_v2

Logger v2

aanchan committed:
Initial commit after deleting speech recoognition code

aanchan committed:
Updated schemas and write references

aanchan committed:
Made modifications to work with Minio locally

aanchan committed:
disabled mysql ec2 connections

aanchan committed:
Modified requirements.txt

aanchan committed:
Merge pull request #20 from SlangLab-NU/disable-mysql

Disable mysql

macarious committed:
Update secure variable to boolean

aanchan committed:
Merge pull request #19 from SlangLab-NU/add-minio

Made modifications to work with Minio locally

aanchan committed:
Merge pull request #18 from SlangLab-NU/disable-speech-recognition

Disable speech recognition

aanchan committed:
Modified RecorderMUI.js to get more screen real estate

aanchan committed:
Merge pull request #22 from SlangLab-NU/front-end-prompt-fix

Front end prompt fix

macarious committed:
Add simple docker file for server

macarious committed:
Add SQLite and ffmpeg installation in docker file

macarious committed:
Add minio to docker (in progress)

macarious committed:
Add Dockerfile for web (in progress)

macarious committed:
Make Dockerfile for MinIO running properly

macarious committed:
Clean up tmp in server

Commit:
Merge branch '23-ephemeral-database-tables-vs-permanent-ones' of https://github.com/SlangLab-NU/VoiceCollector into 23-ephemeral-database-tables-vs-permanent-ones

Commit:
update references at app init

macarious committed:
Make a docker-compose but server and web dockerfiles are not working

macarious committed:
Fix server's dockerfile to connect to MinIO properly

macarious committed:
Frontend dockerfile loads only the main page

macarious committed:
Change ENV in Dockerfile.web to work with docker-compose

macarious committed:
Verify working docker-compose file

macarious committed:
Add DOCKER_README.md to provide instructions in running the docker files

macarious committed:
Add code to create database.db when write_references_to_db fails in __init__.py

aanchan committed:
Merge pull request #24 from SlangLab-NU/23-ephemeral-database-tables-vs-permanent-ones

23 ephemeral database tables vs permanent ones

macarious committed:
Fix missing yarn.lock file when running docker file

macarious committed:
Merge pull request #25 from SlangLab-NU/create-dockerfile

Create dockerfile

macarious committed:
Edit Dockerfile.server to run in both dev and prod environments

macarious committed:
Edit logger.py so it runs independent of  where the root directory is

macarious committed:
Edit db_helper.py so it does not rely on working directory

macarious committed:
Update DOCKER_README.md

macarious committed:
Update DOCKER_README.md to show new instructions

macarious committed:
Create dockerfiles directory for organization

macarious committed:
Fix path issues in docker-compose files

macarious committed:
Add instructions in DOCKER_README.md to run and build the docker-compose together

macarious committed:
Add environment variable in docker-compose.yml files for customization

macarious committed:
Move environment variables from server and web in docker-compose.yml files

macarious committed:
Update references.txt to previous version

Commit:
addressed an issue where the server cannot access minio storage

Commit:
update readme to include instructions for updating server hostname

Commit:
covert api path to most likely external url

Commit:
change schema to https

Commit:
change scheme to https

ztybigcat committed:
allow passing of parameters to react build

macarious committed:
Merge pull request #30 from SlangLab-NU/26-change-backend-server-to-gunicorn

26 change backend server to gunicorn

ztybigcat committed:
draft docker image to dockerhub and release manual

ztybigcat committed:
Reconfigure nginx to serve on ssl

ztybigcat committed:
allow initial obrain of certificate

ztybigcat committed:
Fix bugs in configuration

ztybigcat committed:
put minio user data in persistent storage and minor improvements

ztybigcat committed:
change minio mapping path

ztybigcat committed:
update documentation

ztybigcat committed:
Make the documentation more friendly

ztybigcat committed:
minor change

ztybigcat committed:
minor change 2

aanchan commented on README.md:
```diff
@@ -1,7 +1,21 @@
+
+2. **A Treasure Trove of Data**: We're proud to offer a dataset packed with high-quality audio from dysarthria speakers, detailing different severity levels. As of now, this is the most extensive public dataset in this unique domain.
+
+3. **Powered by Cutting-Edge Tech**: We've embraced wav2vec 2.0, a top-notch pre-trained speech recognition model, to refine our validation process. This ensures pristine data quality and also provides additional insights like intelligibility scores.
```
I removed wav2vec2.0 on this @ztybigcat

aanchan replied:
We do not do ASR validation on it, the models were too heavy.

aanchan replied:
I would write a bit more about the layout of our project, sounds dry but what our front-end and backend are in and so on.

ztybigcat replied:
Replaced those content with an overview of different technologies used in the application.

aanchan commented on README.md:
```diff
@@ -1,7 +1,21 @@
-Create your *own branch* to do implemetation.
-After implementation, push the changes to your branch then create a pull request to dev branch.
-DO NOT PUSH TO MAIN!
+### **Why is Our Dataset Special? 🌟**
```
We don't have a dataset yet :D

ztybigcat replied:
Section removed

aanchan commented on server/README.md:
We probably need to remove this README.md

ztybigcat replied:
Removed

aanchan commented on web/README.md:
Like the server README.md, do we need this README.md here?

ztybigcat replied:
Removed

aanchan commented on server/torgo_results.txt:
This can be deleted.

ztybigcat replied:
Removed

aanchan commented on server/tmp/normal.wav:
This I think can be deleted as well.

ztybigcat replied:
Removed

aanchan commented on server/tests/test_convert_format.py:
Not sure if this test file is actually used

ztybigcat replied:
Removed

aanchan commented on server/app.yaml:
```diff
@@ -0,0 +1,15 @@
+# Google App Engine config file
```
Do we really need all these comments here?

ztybigcat replied:
Removed relevant config files for google cloud.

aanchan commented on server/app/scripts/read_audio.py:
This file might actually not be used.

ztybigcat replied:
Removed

aanchan commented on web/public/favicon.ico:
We might need a new favicon

ztybigcat replied:
replaced with happy prime icons

aanchan commented on web/public/logo192.png:
Possibly logos as well, if these are not used, should be deleted.

ztybigcat replied:
replaced with happy prime logos

aanchan commented on web/public/logo512.png:
This also if not used should be deleted.

ztybigcat replied:
replaced with happy prime logos

aanchan commented on web/src/assets/1.jpg:
This can be deleted

ztybigcat replied:
Removed

aanchan commented on web/src/assets/2.jpg:
This can also be deleted.

ztybigcat replied:
Removed

aanchan commented on web/src/assets/3.jpg:
delete

ztybigcat replied:
Removed

aanchan commented on web/src/components/AboutUs.js:
This can certainly be deleted, we do not need an About Us section

ztybigcat replied:
Removed

aanchan commented on web/src/components/ContactUs.js:
We need this to be deleted as well.

ztybigcat replied:
Removed

aanchan commented on web/src/data.js:
I think this can be deleted.

ztybigcat replied:
Removed

aanchan commented on web/src/image.js:
I think this is also not used, can be deleted

ztybigcat replied:
Removed

aanchan commented on web/src/pages/Home.js:
This home page needs to be changed so that we can head directly to the record page

ztybigcat replied:
Now user will directly land at record page.

Commit:
remove unnecessary readme

ztybigcat committed:
remove unnecessary config files and update
icons with happy prime icon

ztybigcat committed:
1. Remove outdated resources;
2. Change the home page of the app to directly record.
3. Fix issues in the dev version of docker-compose

ztybigcat committed:
remove more unneeded files

ztybigcat committed:
remove more unneeded files and comment necessarry files

ztybigcat committed:
remove as much as possible

ztybigcat committed:
Update readme to be more reasonable

ztybigcat committed:
Ensure database.db is stored outside of container so it does not get trashed

ztybigcat committed:
Adding .gitkeep file to maintain directory structure

ztybigcat committed:
Use absolute path to bypass the issue where
prod and dev run from different workdir
